src/pages/Scholarships.jsx

```diff
@@ -93,29 +95,45 @@ function ScholarshipsPage() {
     pruneQueryParam(qParams.MAX_AMOUNT);
   }
 
+  const [loadMoreFn, setLoadMoreFn] = useState(() =>
+    Scholarships.list({ sortField, sortDir })
+  );
+
+  const loadMoreScholarships = useCallback(
+    (scholarshipsList) => {
+      let mounted = true;
+      scholarshipsList
+        .then(({ results, next, empty }) => {
+          if (mounted) {
+            setScholarships((prev) => [
+              ...prev,
+              ...results.filter((s) =>
+                s.data.amount.intersectsRange(minAmount, maxAmount)
+              ),
+            ]);
+          }
+          setLoadMoreFn(next);
+          setDisableMoreBtn(empty);
```

Comment:
In case the component is no longer mounted everything should be inside the `if (mounted)` block. Alternatively instead of surrounding everything with `if (mounted)` you could also just short circuit return on `!mounted`:
```js
          if (!mounted) return;
          setScholarships((prev) => [
            ...prev,
            ...results.filter((s) =>
              s.data.amount.intersectsRange(minAmount, maxAmount)
            ),
          ]);
          setLoadMoreFn(next);
          setDisableMoreBtn(empty);
```